keps/sig-node/4341-Eviction-Manager-disk-usage/README.md

```diff
@@ -0,0 +1,209 @@
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+
+## Summary
+
+Node pressure eviction is the process by which the kubelet proactively terminates pods to
+reclaim resources on nodes. The kubelet monitors resources like memory, disk space, and 
+filesystem inodes on our cluster's nodes. The Kubelet uses various parameters like eviction 
+signals and eviction thresholds to make eviction decisions.
+
+The kubelet report node conditions to reflect that the node is under pressure because hard or 
+soft eviction threshold is met, like node condition of memory pressure is caused when 
+available memory on the node has satisfied an eviction threshold and disk pressure is caused 
+when available disk space has satisfied an eviction threshold.
+
+It seems that under the condition of disk pressure, Eviction-manager checks disk usage of 
+only living containers of a pod, then decide which pod should be evicted. If in the env, there 
+are dead containers that are not running but consuming the disk space then at the time of the 
+disk pressure on node, the Eviction manager will not check the disk usage of those dead 
+containers due to which the living containers are evicted first. In idol case, the Eviction 
+manager should first evict the dead containers consuming disk space so that the living 
+containers could be saved from eviction.
+
+For more information and results see this issue:
+https://github.com/kubernetes/kubernetes/issues/115201
+
+## Motivation
+
+Eviction Manager checks usage of only living containers, there is no problem in Memory 
+Pressure, since memory used by containers are freed up when they are dead but disk space 
+aren't freed up even after they are dead. Due to this the Eviction Manager evicts the wrong pods.
+
+For example there are two Pods, Pod A and Pod B. 
+Pod A has 1 living container whose disk usage is 5 GB
+Pod B has 1 living container whose disk usage is 0 GB and 1 dead container whose disk usage is 10 GB 
+In current scenario, under the condition of disk pressure, Eviction manager attempts to evict 
+Pod A(total disk usage is less than Pod B) prior to Pod B
+
+The expectation of this enhancement is that in disk pressure case, Eviction manager checks 
+disk usage of living and dead containers of a pod and then decide which pod should be evicted (Pod B in above case)
```

Comment:
Should we divide the dead containers type into initial container, general container, ephemeral containers？